tests/avify4mtest.c

```diff
@@ -0,0 +1,177 @@
+            ((plane == AVIF_CHAN_Y) ? image1->width : uvWidth) * ((image1->depth > 8) ? sizeof(uint16_t) : sizeof(uint8_t));
+        const uint32_t height = (plane == AVIF_CHAN_Y) ? image1->height : uvHeight;
+        for (uint32_t y = 0; y < height; ++y) {
+            if (memcmp(image1->yuvPlanes[plane] + y * image1->yuvRowBytes[plane],
+                       image2->yuvPlanes[plane] + y * image2->yuvRowBytes[plane],
+                       pixelByteCount) != 0) {
+                printf("ERROR: different px at row %" PRIu32 ", channel %" PRIu32 "\n", y, plane);
+                return AVIF_FALSE;
+            }
+        }
+    }
+
+    if (image1->alphaPlane != NULL) {
+        if (image2->alphaPlane == NULL || image1->alphaRange != image2->alphaRange ||
+            image1->alphaPremultiplied != image2->alphaPremultiplied) {
+            printf("ERROR: input mismatch\n");
+            return AVIF_FALSE;
+        }
+        const uint32_t pixelByteCount = image1->width * ((image1->depth > 8) ? sizeof(uint16_t) : sizeof(uint8_t));
+        for (uint32_t y = 0; y < image1->height; ++y) {
+            if (memcmp(image1->alphaPlane + y * image1->alphaRowBytes, image2->alphaPlane + y * image2->alphaRowBytes, pixelByteCount) != 0) {
+                printf("ERROR: different px at row %" PRIu32 ", alpha\n", y);
+                return AVIF_FALSE;
+            }
+        }
+    }
+    return AVIF_TRUE;
+}
+
+//------------------------------------------------------------------------------
+
+// Fills each plane of the image with the maximum allowed value.
+static void fillPlanes(avifImage * image)
+{
+    const uint32_t yuvValue = ((image->yuvRange == AVIF_RANGE_LIMITED) ? 235 : 255) << (image->depth - 8);
+    avifPixelFormatInfo info;
+    avifGetPixelFormatInfo(image->yuvFormat, &info);
+    for (int plane = 0; plane < AVIF_PLANE_COUNT_YUV; ++plane) {
+        if (image->yuvPlanes[plane] != NULL) {
+            const uint32_t planeWidth = (plane == AVIF_CHAN_Y) ? image->width
+                                                               : ((image->width + info.chromaShiftX) >> info.chromaShiftX);
+            const uint32_t planeHeight = (plane == AVIF_CHAN_Y) ? image->height
+                                                                : ((image->height + info.chromaShiftY) >> info.chromaShiftY);
+            for (uint32_t y = 0; y < planeHeight; ++y) {
+                uint8_t * const row = image->yuvPlanes[plane] + y * image->yuvRowBytes[plane];
+                if (image->depth == 8) {
+                    memset(row, yuvValue, planeWidth);
+                } else {
+                    for (uint32_t x = 0; x < planeWidth; ++x) {
+                        ((uint16_t *)row)[x] = yuvValue;
+                    }
+                }
+            }
+        }
+    }
+    if (image->alphaPlane != NULL) {
+        const uint32_t alphaValue = ((image->alphaRange == AVIF_RANGE_LIMITED) ? 235 : 255) << (image->depth - 8);
```

Comment:
Limited-range alpha has been removed from the next version of AVIF. So we don't need to test that case.